### PR TITLE
fix: popups wallet

### PR DESCRIPTION
**Pull request by nuanyang233**
closes #

### CODE REVIEW

File: packages/maskbook/src/extension/popups/pages/Wallet/DeleteWallet/index.tsx [hunk 1]
```diff
@@ -70,26 +69,25 @@ const useStyles = makeStyles()({
...
-            const wallets = await WalletRPC.getWallets()
-            if (wallets.length) {
-                await WalletRPC.resetAccount({
-                    account: first(wallets)?.address,
-                })
-            }
+            const wallets = await WalletRPC.getWallets(ProviderType.MaskWallet)
+
+            await WalletRPC.resetAccount({
+                account: wallets.length ? first(wallets)?.address : '',
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Wallet/DeleteWallet/index.tsx:82]:
```suggestion
                account: first(wallets)?.address : '',
```

File: packages/maskbook/src/locales/en-US.json [hunk 2]
```diff
@@ -905,6 +905,10 @@
     "popups_wallet_unlock_wallet": "Unlock Wallet",
     "popups_wallet_unlock_error_password": "Password is incorrect",
     "popups_wallet_activity_to_address": "To: {{address}}",
+    "popups_wallet_transfer_error_tip": "Network connection failure or contract error, send transaction failure.",
+    "popups_wallet_re_send": "Re-Send",
```
Comment (guanbinrui) [inline packages/maskbook/src/locales/en-US.json:909]:
```suggestion
    "popups_wallet_re_send": "Resend",
```

Is this better?

File: packages/maskbook/src/locales/en-US.json [hunk 3]
```diff
@@ -905,6 +905,10 @@
     "popups_wallet_unlock_wallet": "Unlock Wallet",
     "popups_wallet_unlock_error_password": "Password is incorrect",
     "popups_wallet_activity_to_address": "To: {{address}}",
+    "popups_wallet_transfer_error_tip": "Network connection failure or contract error, send transaction failure.",
```
Comment (guanbinrui) [inline packages/maskbook/src/locales/en-US.json:908]:
```suggestion
    "popups_wallet_transfer_error_tip": "Failed to send transaction due to network connection failure or contract error.",
```

File: packages/maskbook/src/plugins/Wallet/services/rpc.ts [hunk 4]
```diff
@@ -69,6 +73,32 @@ export async function pushUnconfirmedRequest(payload: JsonRpcPayload) {
...
+
+    if (!chunk_?.requests.length) throw new Error('No requests to updated')
+
+    const requests =
+        chunk_?.requests?.map((item) => {
+            if (item.id !== payload.id) return item
+            return payload
+        }) ?? []
+
+    console.log(requests)
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/rpc.ts:90]:
```suggestion
```

File: packages/dashboard/src/pages/Labs/index.tsx [hunk 5]
```diff
@@ -60,10 +60,11 @@ export default function Plugins() {
 
     const account = useAccount()
     const { setDialog: setBuyDialog } = useRemoteControlledDialog(PluginMessages.Transak.buyTokenDialogUpdated)
-    const openTransakDialog = useCallback(() => {
+    const openTransakDialog = useCallback((code?: string) => {
         setBuyDialog({
             open: true,
             address: account,
+            code,
```
[the remark this answers is not in the document]
Comment (nuanyang233) [reply packages/dashboard/src/pages/Labs/index.tsx:67]: The `code` comes from URL params. Just symbol code

File: packages/maskbook/src/extension/popups/pages/Wallet/GasSetting/GasSetting1559.tsx [hunk 6]
```diff
@@ -222,42 +215,67 @@ export const GasSetting1559 = memo(() => {
...
+        if (new BigNumber(maxFeePerGas).isLessThan(gasNow?.slow?.suggestedMaxFeePerGas ?? 0))
+            return t('wallet_transfer_error_max_fee_too_low')
+        if (
+            new BigNumber(maxFeePerGas).isGreaterThan(
+                new BigNumber(gasNow?.fast?.suggestedMaxFeePerGas ?? 0).multipliedBy(HIGH_FEE_WARNING_MULTIPLIER),
+            )
+        )
+            return t('wallet_transfer_error_max_fee_too_high')
+        return undefined
+    }, [maxFeePerGas, gasNow])
```
[the remark this answers is not in the document]
Comment (nuanyang233) [reply packages/maskbook/src/extension/popups/pages/Wallet/GasSetting/GasSetting1559.tsx:278]: Just named `gasnow`. The data source from other

File: packages/maskbook/src/locales/en-US.json (hunk 2 above)
Comment (nuanyang233) [reply packages/maskbook/src/locales/en-US.json:909]: Designer's decision

File: packages/maskbook/src/plugins/Wallet/services/rpc.ts [hunk 7]
```diff
@@ -69,6 +73,31 @@ export async function pushUnconfirmedRequest(payload: JsonRpcPayload) {
     return payload
 }
 
+export async function updateUnconfirmedRequest(payload: JsonRpcPayload) {
+    const now = new Date()
+    const t = createTransaction(await createWalletDBAccess(), 'readwrite')('UnconfirmedRequestChunk')
+
+    const chunk_ = await t.objectStore('UnconfirmedRequestChunk').get(MAIN_RECORD_ID)
+
+    if (!chunk_?.requests.length) throw new Error('No requests to updated')
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/rpc.ts:82]:
```suggestion
    if (!chunk_?.requests.length) throw new Error('No request to updated.')
```

File: packages/maskbook/src/plugins/Wallet/services/rpc.ts [hunk 8]
```diff
@@ -69,6 +73,31 @@ export async function pushUnconfirmedRequest(payload: JsonRpcPayload) {
...
+
+    const requests =
+        chunk_?.requests?.map((item) => {
+            if (item.id !== payload.id) return item
+            return payload
+        }) ?? []
+
+    const chunk = {
+        ...chunk_,
+        updateAt: now,
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/rpc.ts:92]:
```suggestion
        updatedAt: now,
```

File: packages/maskbook/src/extension/popups/components/NetworkSelector/index.tsx [hunk 9]
```diff
@@ -52,18 +52,23 @@ const useStyles = makeStyles()((theme) => ({
 }))
 
 export const NetworkSelector = memo(() => {
-    const currentChainId = useValueRef(currentChainIdSettings)
-    const account = useAccount(ProviderType.MaskWallet)
+    const account = useAccount()
+    const currentChainId = useValueRef(currentMaskWalletChainIdSettings)
+    const currentProvider = useValueRef(currentProviderSettings)
     const { value: networks } = useAsync(async () => WalletRPC.getSupportedNetworks(), [])
+    console.log(currentChainId)
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/components/NetworkSelector/index.tsx:59]:
```suggestion
```

File: packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx [hunk 10]
```diff
@@ -205,9 +214,10 @@ const ContractInteraction = memo(() => {
     const { value: defaultPrices } = useAsync(async () => {
         if (networkType === NetworkType.Ethereum && !maxFeePerGas && !maxPriorityFeePerGas) {
             const response = await WalletRPC.getEstimateGasFees(chainId)
+            // Gwei to wei
             return {
-                maxPriorityFeePerGas: response?.medium.suggestedMaxPriorityFeePerGas ?? 0,
-                maxFeePerGas: response?.medium.suggestedMaxFeePerGas ?? 0,
+                maxPriorityFeePerGas: formatGweiToWei(response?.medium.suggestedMaxPriorityFeePerGas ?? 0).toString(16),
+                maxFeePerGas: formatGweiToWei(response?.medium.suggestedMaxFeePerGas ?? 0).toString(16),
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx:220]:
```suggestion
                maxPriorityFeePerGas: `0x${formatGweiToWei(response?.medium.suggestedMaxPriorityFeePerGas ?? 0).toString(16)}`,
                maxFeePerGas: `0x${formatGweiToWei(response?.medium.suggestedMaxFeePerGas ?? 0).toString(16)}`,
```

File: packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx [hunk 11]
```diff
@@ -223,21 +233,26 @@ const ContractInteraction = memo(() => {
...
-            const toBeClose = new URLSearchParams(location.search).get('toBeClose')
-            await Services.Ethereum.confirmRequest(request.payload)
-
-            if (toBeClose) {
-                window.close()
-            } else {
+            try {
+                await Services.Ethereum.confirmRequest(request.payload)
                 history.goBack()
+            } catch (error_) {
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx:239]:
```suggestion
            } catch {
```

File: packages/maskbook/src/extension/popups/pages/Wallet/GasSetting/GasSetting1559.tsx [hunk 12]
```diff
@@ -209,8 +216,14 @@ export const GasSetting1559 = memo(() => {
...
-                setValue('maxPriorityFeePerGas', value.computedPayload._tx.maxPriorityFeePerGas)
-                setValue('maxFeePerGas', value.computedPayload._tx.maxFeePerGas)
+                setValue(
+                    'maxPriorityFeePerGas',
+                    formatWeiToGwei(new BigNumber(value.computedPayload._tx.maxPriorityFeePerGas, 16)).toString(),
+                )
+                setValue(
+                    'maxFeePerGas',
+                    formatWeiToGwei(new BigNumber(value.computedPayload._tx.maxFeePerGas, 16)).toString(),
+                )
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Wallet/GasSetting/GasSetting1559.tsx:226]:
```suggestion
                setValue(
                    'maxPriorityFeePerGas',
                    `0x${formatWeiToGwei(new BigNumber(value.computedPayload._tx.maxPriorityFeePerGas, 16)).toString()}`,
                )
                setValue(
                    'maxFeePerGas',
                    `0x${formatWeiToGwei(new BigNumber(value.computedPayload._tx.maxFeePerGas, 16)).toString()}`,
                )
```

File: packages/maskbook/src/extension/popups/pages/Wallet/SelectWallet/index.tsx [hunk 13]
```diff
@@ -65,40 +96,18 @@ const useStyles = makeStyles()({
...
-        async (address) => {
-            await WalletRPC.updateAccount({
-                account: address,
-                providerType: ProviderType.MaskWallet,
-            })
-            history.replace(PopupRoutes.Wallet)
-        },
-        [history],
-    )
+    const chainId = Number(search.get('chainId')) as unknown as ChainId
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Wallet/SelectWallet/index.tsx:110]:
```suggestion
    const chainId = Number.parseInt(search.get('chainId'), 10) as unknown as ChainId
```

Comment (guanbinrui) [reply packages/maskbook/src/extension/popups/pages/Wallet/SelectWallet/index.tsx:110]: ATTENTION: the chain id may be altered by the user.

File: packages/maskbook/src/plugins/Wallet/services/rpc.ts [hunk 14]
```diff
@@ -50,7 +51,10 @@ export async function pushUnconfirmedRequest(payload: JsonRpcPayload) {
     const requests = chunk_?.requests ?? []
 
     // validate if it's still possible to push a new request
-    if (requests.length >= MAX_UNCONFIRMED_REQUESTS_SIZE) throw new Error('Unable to add more requests.')
+    if (requests.length >= MAX_UNCONFIRMED_REQUESTS_SIZE) {
+        await openPopupsWindow()
+        throw new Error('Unable to add more requests.')
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/rpc.ts:56]:
```suggestion
        throw new Error('Unable to add more request.')
```

File: packages/maskbook/src/plugins/Wallet/services/account.ts [hunk 15]
```diff
@@ -55,10 +53,20 @@ export async function updateAccount(
...
 
+export async function updateMaskAccount(options: { account?: string; chainId?: ChainId; networkType?: NetworkType }) {
+    if (options.chainId && !options.networkType) options.networkType = getNetworkTypeFromChainId(options.chainId)
+    if (!options.chainId && options.networkType) options.chainId = getChainIdFromNetworkType(options.networkType)
+
+    const { account, chainId, networkType } = options
+
+    if (chainId) currentMaskWalletChainIdSettings.value = chainId
+    if (networkType) currentMaskWalletNetworkSettings.value = networkType
+    if (account) currentAccountMaskWalletSettings.value = account
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/account.ts:67]: `EthereumWallet.isValid(address)`

File: packages/maskbook/src/extension/popups/pages/Wallet/GasSetting/GasSetting1559.tsx (hunk 12 above)
Comment (nuanyang233) [reply packages/maskbook/src/extension/popups/pages/Wallet/GasSetting/GasSetting1559.tsx:226]: Wei to Gwei

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**